pkg/admin/autorecovery/backup.go

```diff
@@ -0,0 +1,56 @@
+package autorecovery
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/openshift/microshift/pkg/admin/data"
+	"github.com/openshift/microshift/pkg/admin/prerun"
+	"github.com/openshift/microshift/pkg/util"
+	"k8s.io/klog/v2"
+)
+
+func GetBackupName() (data.BackupName, error) {
+	isOstree, err := util.PathExists("/run/ostree-booted")
+	if err != nil {
+		return "", fmt.Errorf("failed to check if system is ostree: %w", err)
+	}
+
+	var versionPart string
+	if isOstree {
+		versionPart, err = prerun.GetCurrentDeploymentID()
+		if err != nil {
+			return "", err
+		}
+	} else {
+		execVer, err := prerun.GetVersionOfExecutable()
+		if err != nil {
+			return "", fmt.Errorf("failed to get version of MicroShift executable: %w", err)
+		}
+		versionPart = execVer.String()
+	}
+
+	// Format current time as yyyymmddHHMMSS
+	t := time.Now().Format("20060102150405")
```

Comment:
Out of my ignorance, how does the format "know" which is a month and which is a day?
Should it be `20060130...` to be move explicit?

Reply:
These values are defined in the Go source code and it knows how to search for them: https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/time/format.go;l=57-71

Reply:
Longer answer would be that it has some mapping between meaning and values, like:
01/1 means month, some form of `2` is day, hour is 3/15/03, minute is 4/04, and so on